core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/ldap/ReadWriteLDAPUserStoreManager.java

```diff
@@ -1352,82 +1352,93 @@ public void doUpdateRoleListOfUser(String userName, String[] deletedRoles, Strin
                 // updating the LDAP.
                 for (String deletedRole : deletedRoles) {
 
-                    LDAPRoleContext context = (LDAPRoleContext) createRoleContext(deletedRole);
-                    deletedRole = context.getRoleName();
-                    String searchFilter = context.getSearchFilter();
-
-                    if (isExistingRole(deletedRole)) {
-                        roleSearchFilter = searchFilter.replace("?", escapeSpecialCharactersForFilter(deletedRole));
-                        String[] returningAttributes = new String[]{membershipAttribute};
-                        String searchBase = context.getSearchBase();
-                        NamingEnumeration<SearchResult> groupResults =
-                                searchInGroupBase(roleSearchFilter,
-                                        returningAttributes,
-                                        SearchControls.SUBTREE_SCOPE,
-                                        mainDirContext,
+                    if (StringUtils.isNotEmpty(deletedRole)) {
```

Comment:
I don't this this will be ever empty unless the array includes an empty string like ""

Reply:
Array contains empty strings, it is the root cause for IDENTITY-5024

Reply:
Ok but I think we should check why the deleted roles include empty strings. Is it acceptable as well until it comes to invoking user store API.

Reply:
Error occurs when invoking through admin service.